content/FIDO2/FIDO2_WebAuthn_Developer_Guide/WebAuthn_Server_Registration.adoc

```diff
@@ -0,0 +1,72 @@
+
+== WebAuthn Server Registration
+
+=== Initiate Registration
+To initiate the registration ceremony the application will construct a StartRegistrationOptions instance using its builder and pass that info into the RelyingParty startRegistration method. The only require parameter is a UserIdentity describing the user for which to create a credential. The id field contains the user handle. This should be a stable, unique identifier for the user - equivalent to a username. However due to privacy considerations it is recommended to set the user handle to a random byte array rather than a username.
+
+[source,java]
+----
+byte[] userHandle = new byte[64];
+random.nextBytes(userHandle);
+
+PublicKeyCredentialCreationOptions request = rp.startRegistration(StartRegistrationOptions.builder()
+    .user(UserIdentity.builder()
+        .name("alice")
+        .displayName("Alice Hypothetical")
+        .id(new ByteArray(userHandle))
+        .build())
+    .build());
+----
+
+The startRegistration method returns a PublicKeyCredentialCreationOptions which can be serialized to JSON, sent to the client, and passed as the publicKey argument to navigator.credentials.create(). This request should be stored in temporary storage so that it can later be used in the finish registration step.
```

Comment:
Some of these fields are byte arrays, which I believe will be converted to base64 encoded strings by the JSON serializer. They will however need to be explicitly handled on the client side before they can be passed to `navigator.credentials.create()`.

Reply:
I took a stab at addressing this in eacd866. Let me know if it is not clear enough to proceed for now. I plan to work with Emil to synchronize this section with the library api later.